pyrpl/software_modules/lockbox/models/fabryperot.py

```diff
@@ -104,18 +83,19 @@ def calibrate(self, autosave=False):
         else:
             return None
 
-    def fit_with_voigt(self, curve, voltage):
+    def fit(self, curve, voltage):
         """ Fits curve to a Voigt profile. """
         x_for_fit = self.voltage_to_fit(voltage)
 
         peak = VoigtModel()
-        p_params = peak.guess(curve, x=x_for_fit)
+        p_params = peak.guess(curve, x=x_for_fit, negative=self.negative)
         background = ConstantModel()
         b_params = background.guess(curve)
         background.set_param_hint("c", value=np.min(curve))
         mod = peak + background
         pars = p_params + b_params
-        result = mod.fit(curve, pars, x=x_for_fit, method="least_squares")
+        result = mod.fit(curve, pars, x=x_for_fit, method="least_squares", 
+                         max_nfev=self.max_nfev)
```

Comment:
Nice. I think it's very useful to set a limit here. Somewhat bizarre that the default is `None`.